runtime/src/main/java/org/corfudb/runtime/object/StreamViewSMRAdapter.java

```diff
@@ -98,14 +97,12 @@ public void gc(long trimMark) {
      */
     public List<SMREntry> current() {
         ILogData data = streamView.current();
-        if (data != null) {
-            if (data.getType() == DataType.DATA
-                    && data.getPayload(runtime) instanceof ISMRConsumable) {
-                return ((ISMRConsumable) data.getPayload(runtime))
-                        .getSMRUpdates(streamView.getId());
-            }
+        if (data != null && data.getType() == DataType.DATA
+                && data.getPayload(runtime) instanceof ISMRConsumable) {
+            return ((ISMRConsumable) data.getPayload(runtime))
+                    .getSMRUpdates(streamView.getId());
         }
-        return null;
+        return Collections.emptyList();
```

Comment:
have you checked all the places executing this code? If they have checks for null, we should be very careful here, we can break some of them silently

Reply:
Yes, checked all usages. They are are in current and previous and the change not breaking them.

Reply:
The comment should be updated too.

Reply:
done